### ISSUE TITLE
Consider dropping Twirpy middlewares

### ISSUE THREAD

**Issue by ofpiyush** [verloop/twirpy#42]
One can use middlewares from https://www.starlette.io/middleware/

Our middlewares act more like a decorator on the actual method than a real middleware. I believe we had added them internally when we didn't have hooks, but I can't remember for sure.

To avoid confusion with Starlette's middlewares, we can drop them from a future version.

As for immediate next step, maybe we can deprecate them in the next version and wait for users to tell us if they're using it?

If no one shows up for the next 2 versions, we can drop support?

**Comment by jacobe** [comment]
Good idea. It's confusing and also undocumented, so I had to reverse engineer how to add middlewares to an instance of TwirpASGIApp.
